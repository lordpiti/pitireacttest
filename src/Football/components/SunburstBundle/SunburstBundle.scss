.CDTtooltip {
  background-color: yellow;
  padding: 20px;
  position: absolute;
  text-align: center;
  font: 12px sans-serif;
  border: 0px;
  border-radius: 8px;
  pointer-events: none;
}

.subsLimitSelect {
  // margin-top: 45px;
  margin-left: 45px;
  position: absolute;
}
