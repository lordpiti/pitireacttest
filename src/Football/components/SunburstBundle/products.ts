export type Product = {
  groupDescription?: string | null;
  code: string;
  description: string;
  substituteProducts: SubsProduct[];
  parentId?: number;
  isOrphan: boolean;
  isExcluded?: boolean;
  highlightGradient?: string;
  measure?: number;
};

export type SubsProduct = {
  substituteProductCode: string;
  substituteScore: number;
  relativePairwiseScore: number;
};
