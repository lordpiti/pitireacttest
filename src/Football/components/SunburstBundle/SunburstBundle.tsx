// import { Product } from 'components/TreeNode/TreeNode';
import React from 'react';
import { Bundle } from '../Bundle/Bundle';
import { Sunburst } from '../Sunburst/Sunburst';
import './SunburstBundle.scss';

interface SunburstBundleProps {
  data: any;
  diameter: number;
  selectedProduct?: any;
  setSelectedProduct: Function;
}

interface SunburstBundleState {
  tooltipCssStyle?: any;
  tooltipHidden: boolean;
  tooltipText?: string;
  subsLimit?: number;
}

export class SunburstBundle extends React.Component<
  SunburstBundleProps,
  SunburstBundleState
> {
  state: SunburstBundleState = {
    tooltipHidden: true,
    tooltipCssStyle: {},
    tooltipText: '',
    subsLimit: 10,
  };

  componentDidUpdate(prevProps: SunburstBundleProps) {
    debugger;
    if (prevProps.selectedProduct !== this.props.selectedProduct) {
      this.setSubsLimit(this.props.selectedProduct);
    }
  }

  updateToolTip = (element: string, d: any) => {
    const popupState: SunburstBundleState = {
      tooltipHidden: true,
    };

    if (element) {
      popupState.tooltipCssStyle = {
        left: d.nativeEvent.offsetX,
        top: d.nativeEvent.offsetY,
      };
      popupState.tooltipText = element;
      popupState.tooltipHidden = false;
    }

    this.setState(popupState);
  };

  selectProduct = (d, product: any) => {
    const filteredProduct = product && {
      ...product,
    };
    this.props.setSelectedProduct(filteredProduct);
    this.setSubsLimit(filteredProduct);
  };

  setSubsLimit(product: any) {
    let limit = 10;
    if (product) {
      limit = product.substituteProducts.length;
    }

    this.setState({
      // remove : 10 if subslimit should be persisted...
      subsLimit: limit,
    });
  }

  setSubsLimitEventHandler = (e: any) => {
    this.setState({ subsLimit: parseInt(e.target.value, 10) });
  };

  createSelectOption = (i) => (
    <option value={i + 1} selected={this.state.subsLimit === i + 1}>
      {i + 1}
    </option>
  );

  render() {
    const {
      // Destructure any props you will need
      diameter,
      selectedProduct,
    } = this.props;

    const subsArray: any = selectedProduct
      ? selectedProduct.substituteProducts
      : [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

    return (
      <>
        <div className='text-left'>
          Select maximum number of products:
          <select
            className='subsLimitSelect'
            onClick={() => null}
            onChange={this.setSubsLimitEventHandler}
            value={this.state.subsLimit ? this.state.subsLimit : 10}
          >
            {subsArray.map((a, i) => this.createSelectOption(i))}
          </select>
        </div>
        <div id='sunburstBundle' className='text-center'>
          {!this.state.tooltipHidden && (
            <div
              id='tooltipSunburst'
              className='CDTtooltip'
              style={this.state.tooltipCssStyle}
            >
              {this.state.tooltipText}
            </div>
          )}
          <svg
            width={diameter}
            height={diameter}
            onClick={(e) => this.props.setSelectedProduct(null)}
          >
            <g transform={`translate(${diameter / 2},${diameter / 2})`}>
              <Sunburst
                data={this.props.data}
                diameter={diameter}
                updateToolTip={this.updateToolTip}
              />
              <Bundle
                data={this.props.data}
                diameter={diameter}
                selectedProduct={this.props.selectedProduct}
                selectProduct={this.props.setSelectedProduct}
                subsLimit={this.state.subsLimit}
              />
            </g>
          </svg>
        </div>
      </>
    );
  }
}
