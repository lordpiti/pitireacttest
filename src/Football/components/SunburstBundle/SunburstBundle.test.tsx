import { shallow } from 'enzyme';
import { any } from 'prop-types';
import React from 'react';
import { SunburstBundle } from './SunburstBundle';

describe('SunburstBundle', () => {
  const testProps = {
    data: any,
    diameter: 100,
    selectedProduct: null,
    setSelectedProduct: jest.fn()
  };
  const sunburstBundle = shallow(<SunburstBundle {...testProps} />);

  const instance = sunburstBundle.instance() as SunburstBundle;

  it('should render correctly', () => {
    expect(sunburstBundle).toMatchSnapshot();
  });

  describe('updateToolTip', () => {
    it('should update the state', () => {
      const domElement = {
        nativeEvent: {
          offsetX: 10,
          offsetY: 10
        }
      };
      instance.updateToolTip('test', domElement);

      const expectedState = {
        tooltipCssStyle: {
          left: 10,
          top: 10
        },
        subsLimit: 10,
        tooltipText: 'test',
        tooltipHidden: false
      };

      expect(instance.state).toEqual(expectedState);
    });
  });
});
