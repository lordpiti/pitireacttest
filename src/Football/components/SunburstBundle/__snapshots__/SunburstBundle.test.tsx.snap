// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SunburstBundle should render correctly 1`] = `
<Fragment>
  <select
    className="subsLimitSelect"
    onChange={[Function]}
    onClick={[Function]}
    value={10}
  >
    <option
      selected={false}
      value={1}
    >
      1
    </option>
    <option
      selected={false}
      value={2}
    >
      2
    </option>
    <option
      selected={false}
      value={3}
    >
      3
    </option>
    <option
      selected={false}
      value={4}
    >
      4
    </option>
    <option
      selected={false}
      value={5}
    >
      5
    </option>
    <option
      selected={false}
      value={6}
    >
      6
    </option>
    <option
      selected={false}
      value={7}
    >
      7
    </option>
    <option
      selected={false}
      value={8}
    >
      8
    </option>
    <option
      selected={false}
      value={9}
    >
      9
    </option>
    <option
      selected={true}
      value={10}
    >
      10
    </option>
    }
  </select>
  <div
    className="text-center"
    id="sunburstBundle"
  >
    <svg
      height={100}
      onClick={[Function]}
      width={100}
    >
      <g
        transform="translate(50,50)"
      >
        <Sunburst
          data={[Function]}
          diameter={100}
          updateToolTip={[Function]}
        />
        <Bundle
          data={[Function]}
          diameter={100}
          selectProduct={[MockFunction]}
          selectedProduct={null}
          subsLimit={10}
        />
      </g>
    </svg>
  </div>
</Fragment>
`;
